.github/workflows/test-deploy.yml

```
@@ -0,0 +1,132 @@
+      'test-external' || 'test-internal' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - uses: actions/setup-python@v2
+      - run: pip install tensorboard
+      - run: npm ci
+      - run: npm run test
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          TEST_GITHUB_TOKEN: ${{ secrets.TEST_GITHUB_TOKEN }}
+          TEST_GITHUB_REPO: https://github.com/iterative/cml_qa_tests_dummy
+          TEST_GITHUB_SHA: 62edc8b3f46a60b3fe1e5c08fd3e0046d350ee29
+          TEST_GITLAB_TOKEN: ${{ secrets.TEST_GITLAB_TOKEN }}
+          TEST_GITLAB_REPO: https://gitlab.com/iterative.ai/cml_qa_tests_dummy
+          TEST_GITLAB_SHA: c4c13286e78dc252dd2611f31a755f10d343fbd4
+          TEST_BBCLOUD_TOKEN: ${{ secrets.TEST_BBCLOUD_TOKEN }}
+          TEST_BBCLOUD_REPO: https://bitbucket.org/iterative-ai/cml-qa-tests-dummy
+          TEST_BBCLOUD_SHA: 9bb9131ce0af294fe1c6eedca1f2bce3983e80bd
+  packages:
+    needs: [lint, test]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          registry-url: https://registry.npmjs.org
+      - run: npm install
+      - run:
+          npm ${{ github.event_name == 'release' && 'publish' || 'publish
+          --dry-run' }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+  images:
+    runs-on: ubuntu-latest
+    needs: packages
+    strategy:
+      matrix:
+        dvc: [1, 2]
+        base: [0, 1]
+        gpu: [false, true]
+        include:
+          - base: 0
+            ubuntu: 18.04
+            python: 2.7
+            cuda: 10.1
+            cudnn: 7
+          - base: 1
+            ubuntu: 20.04
+            python: 3.8
+            cuda: 11.0.3
+            cudnn: 8
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Metadata
+        id: metadata
+        run: |
+          latest_tag=$(git describe --tags | cut -d- -f1)
+          cml_version=${latest_tag##v}
+          echo ::set-output name=cml_version::$cml_version
+          if [[ ${{ matrix.gpu }} == true ]]; then
+            echo ::set-output name=base::nvidia/cuda:${{ matrix.cuda }}-cudnn${{ matrix.cudnn }}-runtime-ubuntu${{ matrix.ubuntu }}
+            echo ::set-output name=tag::${cml_version//.*/}-dvc${{ matrix.dvc }}-base${{ matrix.base }}-gpu
+          else
+            echo ::set-output name=base::ubuntu:${{ matrix.ubuntu }}
+            echo ::set-output name=tag::${cml_version//.*/}-dvc${{ matrix.dvc }}-base${{ matrix.base }}
+          fi
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key:
+            ${{ runner.os }}-buildx-${{ steps.metadata.outputs.cml_version
+            }}-${{ github.sha }}
+          restore-keys:
+            ${{ runner.os }}-buildx-${{ steps.metadata.outputs.cml_version }}-
```

I wonder if this should be scoped to the deps versions too...

```suggestion
            ${{ runner.os }}-buildx-${{ steps.metadata.outputs.cml_version
            }}-${{ steps.metadata.outputs.tag }}-${{ github.sha }}
          restore-keys:
            ${{ runner.os }}-buildx-${{ steps.metadata.outputs.cml_version
            }}-${{ steps.metadata.outputs.tag }}-
```

---

resolved by https://github.com/iterative/cml/pull/530#discussion_r633497553